### PR TITLE
Core UI Support

### PR DESCRIPTION
**Pull request by MightyAx**
Implementation to resolve [ingest-central#628](https://github.com/HumanCellAtlas/ingest-central/issues/628) & [ingest-central#589](https://github.com/HumanCellAtlas/ingest-central/issues/589)

### CODE REVIEW

File: src/main/java/org/humancellatlas/ingest/project/ProjectRepositoryImpl.java [hunk 1]
```diff
@@ -68,17 +68,20 @@
             criterias.add(criteria);
         }
 
-        query.addCriteria(
-            new Criteria().orOperator(criterias.toArray(
-                new Criteria[criterias.size()]
-            ))
-        );
+        new Criteria();
```
Comment (prabh-t) [inline src/main/java/org/humancellatlas/ingest/project/ProjectRepositoryImpl.java]: think this shouldn't be here?

File: src/main/java/org/humancellatlas/ingest/project/ProjectService.java [hunk 2]
```diff
@@ -60,7 +64,17 @@ public Project addProjectToSubmissionEnvelope(SubmissionEnvelope submissionEnvel
...
-    public Page<Project> queryByContent(List<MetadataCriteria> query, Pageable pageable){
-        return this.projectRepository.findByContent(query, pageable);
+    public Page<Project> queryByContent(List<MetadataCriteria> query, Optional<Boolean> isUpdate, Pageable pageable){
+        return this.projectRepository.findByContent(query, isUpdate, pageable);
+    }
+
+    public Page<SubmissionEnvelope> getProjectSubmissionEnvelopes(Project project, Pageable pageable) {
+        Page<Project> projects = this.projectRepository.findByUuid(project.getUuid(), Pageable.unpaged());
+        List<SubmissionEnvelope> envelopes = new ArrayList<>();
+        for (Project projectDocument : projects)
```
Comment (prabh-t) [inline src/main/java/org/humancellatlas/ingest/project/ProjectService.java]: findByUuid is unlikely to return multi page output in most cases, but still we should maybe not assume that and make sure all projects are iterated over, not just in this page. stream?

File: src/main/java/org/humancellatlas/ingest/project/ProjectRepositoryImpl.java [hunk 3]
```diff
@@ -14,14 +13,15 @@
...
+import java.util.Optional;
 
 public class ProjectRepositoryImpl implements ProjectRepositoryCustom {
 
     @Autowired
     private MongoTemplate mongoTemplate;
 
     @Override
-    public Page<Project> findByContent(List<MetadataCriteria> metadataQuery, Pageable pageable) {
+    public Page<Project> findByContent(List<MetadataCriteria> metadataQuery, Optional<Boolean> isUpdate, Pageable pageable) {
```
Comment (aaclan-ebi) [inline src/main/java/org/humancellatlas/ingest/project/ProjectRepositoryImpl.java]: should we just allow users to query by document fields and not just content fields?

Comment (MightyAx) [reply src/main/java/org/humancellatlas/ingest/project/ProjectRepositoryImpl.java]: That is a better long term solution. Will look into it.

**Comment by MightyAx** [comment]
Closing this as I will make a new branch onto of current master

**Comment by MightyAx** [comment]
Survived by #99

**Pull request closed without merging**